wire-ios-request-strategy/Sources/Message Sending/SessionEstablisher.swift

```diff
@@ -0,0 +1,134 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public enum SessionEstablisherError: Error {
+    case failedToGenerateRequest
+    case failedToParseResponse
+}
+
+public class SessionEstablisher {
+
+    public init(
+        httpClient: HttpClient,
+        context: NSManagedObjectContext
+    ) {
+        self.httpClient = httpClient
+        self.managedObjectContext = context
+    }
+
+    private let httpClient: HttpClient
+    private let managedObjectContext: NSManagedObjectContext
+    private let requestFactory = MissingClientsRequestFactory()
+    private let processor = PrekeyPayloadProcessor()
+
+    func establishSession(with clients: Set<UserClient>, apiVersion: APIVersion) async -> Swift.Result<Void, SessionEstablisherError> {
+
+        // Establish sessions in chunks and return on first error
+        for chunk in Array(clients).chunked(into: 28) {
```

Comment:
use constant, why 28?

Reply:
Took the constant from the existing implementation. I moved it to a constant in a follow up PR